### PR TITLE
Add Builder Design Pattern

### PR DESCRIPTION
**Pull request by aquaruiz**

### CODE REVIEW

**Comment by aquaruiz** [comment]
@qnislavnachev

File: design-patterns/creational-patterns/src/builder/StoneHouseBuilder.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
+package builder;
+
+public class StoneHouseBuilder implements HouseBuilder {
+	private House result;
```
Comment (qnislavnachev) [inline design-patterns/creational-patterns/src/builder/StoneHouseBuilder.java]: the name of the reference should be 'house' or something similar

File: design-patterns/creational-patterns/src/builder/WoodenCabinBuilder.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
+package builder;
+
+public class WoodenCabinBuilder implements HouseBuilder {
+	private House result;
```
Comment (qnislavnachev) [inline design-patterns/creational-patterns/src/builder/WoodenCabinBuilder.java]: Same here.

**Review by qnislavnachev — CHANGES_REQUESTED** [review]

File: design-patterns/creational-patterns/src/builder/StoneHouseBuilder.java (hunk 1 above)
Comment (aquaruiz) [reply design-patterns/creational-patterns/src/builder/StoneHouseBuilder.java]: Renamed.

File: design-patterns/creational-patterns/src/builder/WoodenCabinBuilder.java (hunk 2 above)
Comment (aquaruiz) [reply design-patterns/creational-patterns/src/builder/WoodenCabinBuilder.java]: Renamed.

**Review by qnislavnachev — APPROVED** [review]

**Pull request merged**